From d51d99c4ec5a448e4ef70ae3cd4fe21d6cd8eb3f Mon Sep 17 00:00:00 2001
From: Bill Nottingham <notting@redhat.com>
Date: Fri, 6 Jan 2012 14:47:04 -0500
Subject: [PATCH 105/109] Handle dmraid sets with spaces (#728795,
 <lnykryn@redhat.com>) (cherry picked from commit
 7414b9457fba61af86b0756a37bc994d079bc621)

---
 systemd/fedora-storage-init |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/systemd/fedora-storage-init b/systemd/fedora-storage-init
index 66ba6a8..565d811 100755
--- a/systemd/fedora-storage-init
+++ b/systemd/fedora-storage-init
@@ -19,6 +19,8 @@ if ! strstr "$cmdline" nodmraid && [ -x /sbin/dmraid ]; then
 	modprobe dm-mirror >/dev/null 2>&1
 	dmraidsets=$(LC_ALL=C /sbin/dmraid -s -c -i)
 	if [ "$?" = "0" ]; then
+		SAVEIFS=$IFS
+		IFS=$(echo -en "\n\b")
 		for dmname in $dmraidsets; do
 			if [[ "$dmname" == isw_* ]] && \
 			   ! strstr "$cmdline" noiswmd; then
@@ -27,6 +29,7 @@ if ! strstr "$cmdline" nodmraid && [ -x /sbin/dmraid ]; then
 			/sbin/dmraid -ay -i --rm_partitions -p "$dmname" >/dev/null 2>&1
 			/sbin/kpartx -u -a -p p "/dev/mapper/$dmname"
 		done
+		IFS=$SAVEIFS
 	fi
 fi
 
-- 
1.7.9.2

